portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,34 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+private ArrayList<String> messages = new ArrayList<>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    response.getWriter().println("<h1>Hello Shengjing Zhang!</h1>");
+
+    Gson gson = new Gson();
+    String json = gson.toJson(messages);
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
```

Comment:
Semi colon might be typo https://stackoverflow.com/questions/477816/what-is-the-correct-json-content-type